### ISSUE TITLE
[2020 arXiv] Improved Baselines with Momentum Contrastive Learning

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#58]
MoCo v2 논문

SimCLR 논문에서의 몇가지 다른 점들을 들고와서 실험을 통해 어떤 점들이 효과가 있었는지 ablation study 느낌으로 보여줌
(SimCLR와 실험 세팅을 비슷하게 맞춰놓고 어떤 알고리즘이 더 우수한지 비교하는 느낌?)

## 1. MoCo
<img width="971" alt="image" src="https://user-images.githubusercontent.com/49643709/159412741-670d7843-d4d4-45b8-9097-147d444d4286.png">

MoCo는 SimCLR 같은 end-to-end 구조가 아니라 queue에 negative sample들을 저장해놓고 이를 이용하여 contrastive learning하는 것임

Large batch를 필요하지 않는다는 점에서 computation cost가 end-to-end 구조보다 훨씬 적음

## 2. Experiments

1.28M ImageNet으로 학습하여 2가지 dataset에서 test

1. ImageNet linear classification : encoder freeze하고 linear classifier만 학습시켜 classification 성능 측정 (linear probing)
2. Transferring to VOC object detection : Faster R-CNN 구조에 넣어 VOC 07+12 trainval로 end-to-end finetune하여 VOC 07 test에서 성능 측정

<img width="668" alt="image" src="https://user-images.githubusercontent.com/49643709/159414449-63719d9d-9ada-4823-9b24-2f03d796c2de.png">

### 2.1. MLP head
<img width="646" alt="image" src="https://user-images.githubusercontent.com/49643709/159414498-3c9d7bd0-b54c-4ffb-8f2f-910f5e29959d.png">

기존의 MoCo v1에서는 fc head를 사용했는데, 이를 2-layer MLP head로 교체
(2048 dimension hidden layer with ReLU)

그리고 SimCLR에서 나왔듯이 optimal temperature tau값을 search하여 최적의 tau값을 찾음

ImageNet에서 성능이 크게 올랐는데 비해 detection task에서의 성능 gain은 그렇게 크지 않았음

### 2.2. Augmentation

SimCLR에서 많은 augmentation을 사용했는데, 그 중 stronger color distortion augmentation을 사용하면 성능 감소가 있었다고 함

그래서 blur augmentation만 추가해서 실험 진행

Extra augmentation을 사용하면 MLP head를 사용했을 때에 비해 ImageNet 성능 gain은 작지만 detection에서의 성능 gain은 큼

이를 통해 linear classifier accuracy가 transfer performance in detection과 monotonically related되지 않다고 볼 수 있음

### 2.3. Comparison with SimCLR
<img width="672" alt="image" src="https://user-images.githubusercontent.com/49643709/159415238-f50524c6-8ccd-48e7-a218-975a3c74f295.png">

SimCLR와 fair comparison을 위해 cosine learning rate scheduler로 학습
(half-period cosine)

MoCo v2가 더 성능이 좋음

### 2.4. Computational cost
<img width="661" alt="image" src="https://user-images.githubusercontent.com/49643709/159415457-27493c14-0b1d-49f3-a60c-da5ae02abbdd.png">

SimCLR 같은 end-to-end case는 GPU cost가 너무 큼
